pkgs/development/tools/yourkit-java/default.nix

```diff
@@ -0,0 +1,39 @@
+{ fetchzip
+, lib
+, stdenv
+, jdk17
+}:
+let jre = jdk17;
+    vPath = v: lib.elemAt (lib.splitString "-" v) 0;
+    arch = if stdenv.targetPlatform.system == "x86_64-linux"
+           then "x64"
+           else if stdenv.targetPlatform.system == "aarch64-linux"
+           then "arm64"
+           else throw "Unsupported system";
+in
+stdenv.mkDerivation rec {
+  pname = "YourKit-JavaProfiler";
+  version = "2023.9-b103";
+  src = fetchzip {
+    url = "https://download.yourkit.com/yjp/${vPath version}/${pname}-${version}-${arch}.zip";
+    hash = "sha256-fJk39cQEU924FViCwTcISIyhiwJEviVeqxLiNQifRis=";
+  };
+
+  buildInputs = [ jre ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    sed -i -e 's|JAVA_EXE="$YD/jre64/bin/java"|JAVA_EXE=${jdk17}/bin/java|' bin/profiler.sh
```

Comment:
Put this line on postPatch.

Reply:
I am not so sure if this is a patch. This change is only necessary for the installation. So is the drop of `jre64` directory in the unpacked files: the original files are perfectly fine; we modify them only to fit into Nix's ecosystem.